rotkehlchen/rotkehlchen.py

```diff
@@ -168,6 +169,14 @@ def __init__(self, args: argparse.Namespace) -> None:
         self.shutdown_event = gevent.event.Event()
         self.migration_manager = DataMigrationManager(self)
 
+    def kill_running_tx_query_tasks(self) -> None:
+        """Attempts to kill greenlets tasks querying transactions."""
+        log.debug(f'Attempting to delete running transaction queries tasks from {self.api_task_greenlets}')  # noqa: E501
+        for task in self.api_task_greenlets:
+            if task.is_txn_query is True:
+                task.greenlet.kill()
+                self.api_task_greenlets.remove(task)
```

Comment:
Doesn't it break because of trying to delete an element from the list that is being iterated?

Reply:
What about this?